### PR TITLE
feat: CLI supports Projects [DET-6820]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Adds CLI commands which call the Python API bindings for Projects.

**This merges into the workspace/projects feature branch**

## Test Plan

can use `det project help` or `det p help`
`det p list Uncategorized` - the default workspace ("Uncategorized") will already be set up on your db and have a default project (also named "Uncategorized")
`det p list-experiments Uncategorized Uncategorized` - should list all experiments in your db
`det p create Uncategorized Added` - creates a new project inside Uncategorized workspace
`det p describe Uncategorized Added` - prints the new project row and an (empty) table of experiments
`det p list Uncategorized --sort-by name` - order should be Added, Uncategorized
`det p edit Uncategorized Added --new_name UpdatedName --description Example` - updates project
`det p delete Uncategorized UpdatedName` - given y/n prompt, deletes project

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by trentwatt — CHANGES_REQUESTED** [review]
Coming together nicely!

Main thing would just be handling args.all, and either filtering current user by default or adjusting help text for `--all` to reflect that we don't. 

pretty much everything else i consider optional.

**Comment by trentwatt** [comment]
Also, I'm not sure if this test plan will even show up in release party since it's getting merged into a feature branch, but it would be nice to flesh it out a bit if so.

On a related note, I think it could use a good bit more automated tests before getting merged into master (different users, moving experiments, etc.), but I don't think it's necessary for now.

File: e2e_tests/tests/cluster/test_workspace_org.py
Comment (mapmeld) [reply e2e_tests/tests/cluster/test_workspace_org.py:177]: yeah this is a good question, I probably should open a thread on this,
- when I delete the test project at the end of testing, it gets blocked by this experiment having it as an association
- when I delete the test experiments before the projects, it gets blocked even though when I manually check the database, the experiment was deleted -- I believe there is a delay before the experiment is erased?

So when I move the test experiment back to the default_project then I don't need to worry about it

File: e2e_tests/tests/cluster/test_workspace_org.py
Comment (mapmeld) [reply e2e_tests/tests/cluster/test_workspace_org.py:93]: I've added a lot of these rules through this PR: https://github.com/determined-ai/determined/pull/3878/files

File: master/internal/api_project.go
Comment (mapmeld) [reply master/internal/api_project.go:196]: ✅  will fix

File: harness/determined/cli/project.py
Comment (mapmeld) [reply harness/determined/cli/project.py:123]: I'll include the project name in the warning (and do the same for workspace name)

File: proto/src/determined/project/v1/project.proto
Comment (mapmeld) [reply proto/src/determined/project/v1/project.proto:13]: In the DB the notes are stored as a JSON array; this is a good point that I don't know how we'll handle update/delete.  The ERD has PutProjectNotes which would either append to or replace the array.

File: master/internal/api_experiment.go
Comment (mapmeld) [reply master/internal/api_experiment.go:1308]: That's correct. For now all experiments are created in the default project.

**Comment by mapmeld** [comment]
@trentwatt  Thanks for looking over a bunch of this code. I've rebased so it should be reduced to the CLI-specific code now

File: harness/determined/cli/project.py
Comment (mapmeld) [reply harness/determined/cli/project.py:254]: added `--all` functionality here ✅

File: harness/determined/cli/project.py
Comment (trentwatt) [inline harness/determined/cli/project.py:114]: super minor nit: with the logic here being the same as in list_project_experiments, would have preferred the code to read the same.

**Review by trentwatt — APPROVED** [review]
Changes look good!

File: harness/determined/cli/project.py
Comment (mapmeld) [reply harness/determined/cli/project.py]: ✅ figured out an issue with kwargs here

**Pull request merged**
